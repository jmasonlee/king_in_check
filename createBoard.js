const files = "abcdefgh".split('')

const getFileIndex = file => files.indexOf(file)
const getRankIndex = rank => rank - 1

export default pieces => {
  const board = Array(8)
    .fill("")
    .map((rank) => {
      return Array(8).fill("");
    });
  
  pieces.forEach(piece => {
    board[getRankIndex(piece.rank)][getFileIndex(piece.file)] = piece.type
  });
  return board;
};