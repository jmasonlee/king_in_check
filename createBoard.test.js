import createBoard from './createBoard'
describe("create board", () => {
  it("should return an empty board when no pieces are passed", () => {
    expect(createBoard([])).toMatchSnapshot();
  });

  it('should return a board with pieces in their location', () => {
    expect(createBoard([{type: 'K', rank: '2', file:'c'}])).toMatchSnapshot()
  });
});
