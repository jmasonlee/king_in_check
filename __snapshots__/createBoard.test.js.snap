// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create board should return a board with pieces in their location 1`] = `
Array [
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "K",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
]
`;

exports[`create board should return an empty board when no pieces are passed 1`] = `
Array [
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
]
`;
